src/lib/nav/NavigationRailItem.svelte

```diff
@@ -0,0 +1,175 @@
+<script lang="ts">
+  import type { IconifyIcon } from "@iconify/types";
+
+  import Layer from "$lib/misc/Layer.svelte";
+  import Icon from "$lib/misc/_icon.svelte";
+
+  let {
+    label,
+    icon,
+    active = false,
+    badge,
+    onclick,
+  }: {
+    label: string;
+    icon: IconifyIcon;
+    active?: boolean;
+    badge?:
+      | {
+          label?: string;
+          small?: boolean;
+        }
+      | true;
+    onclick?: (e: MouseEvent) => void;
+  } = $props();
+</script>
+
+<button class="m3-container" class:active {onclick}>
```

Comment:
only button? no `<a>` upon `href`?

Reply:
putting `button` inside `a` is problematic actually
<img width="1158" height="1047" alt="image" src="https://github.com/user-attachments/assets/f217f891-680f-42bb-be2b-0d4345efd96a" />